scripts/update_lmod_caches.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+function echo_green() {
+    echo -e "\e[32m$1\e[0m"
+}
+
+function echo_red() {
+    echo -e "\e[31m$1\e[0m"
+}
+
+function error() {
+    echo_red "ERROR: $1" >&2
+    exit 1
+}
+
+# Check if a stack base dir has been specified
+if [ "$#" -ne 1 ]; then
+    error "usage: $0 <path to main directory of an EESSI stack>"
+fi
+
+stack_base_dir="$1"
+
+# Check if the given stack base dir exists
+if [ ! -d ${stack_base_dir} ]
+then
+  error "${stack_base_dir} does not point to an existing directory!"
+fi
+
+# Check if Lmod's cache update script can be found at the expected location (in the compatibility layer of the gien stack)
+update_lmod_system_cache_files="${stack_base_dir}/compat/linux/$(uname -m)/usr/share/Lmod/libexec/update_lmod_system_cache_files"
+if [ ! -f ${update_lmod_system_cache_files} ]
+then
+  error "expected to find Lmod's cache update script at ${update_lmod_system_cache_files}, but it doesn't exist."
+fi
+
+# Find all subtrees of supported CPU targets by looking for "modules" directories, and taking their parent directory
+architectures=$(find ${stack_base_dir}/software/ -maxdepth 5 -type d -name modules -exec dirname {} \;)
+
+# For every subtree:
+# - create an .lmod directory;
+# - add an lmodrc.lua file that defines the location of the cache;
```

Comment:
@bedroge Is this part necessary?

There should be an `lmodrc.lua` file in place already, which was created via `create_lmodrc.py` in `software-layer` repo?

I guess it doesn't hurt to create a minimal one if none was found, but I don't see how that could happen?

Reply:
I took some of this from our internal "create tarball" script, which was based on an earlier version of the EESSI one. I now see that we moved this to the EESSI installation script in the software layer. So I agree we can remove this part from here, and then just remove the Lmod cache update from our installation script (and leave the lmodrc stuff there, also becase it does some GPU stuff there as well now).

Reply:
fixed in https://github.com/EESSI/filesystem-layer/pull/175/commits/09aa76b94207db9b7117fc614fb553ad7601375b